src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/special/SpecialWikiActivityPageObject.java

```diff
@@ -65,29 +74,38 @@ public void verifyNewWallThreadEntry(String postTitle, String postContent, Strin
     Assertion.assertNotNull(result);
   }
 
-  /**
-   * verifies if there is the wanted edition on WikiActivity, searching through 5 recent editions.
-   */
-  public void verifyRecentEdition(String articleName, String userName) {
+  public Boolean doesLastNRecentEditionsContain(int n, String articleName, String userName) {
     Boolean ifPassed = false;
-    for (int i = 0; i < 4; i++) {
-      ifPassed = ifDetailsPresent(editActivitiesList.get(i), articleName,
-                                  userName);
-      if (ifPassed) {
-        PageObjectLogging.log("verifyRecentEdition",
-                              "WikiActivity module found recent edition that affected '"
-                              + articleName
-                              + "' and was done by user: " + userName, true);
-        break;
-      }
+    for (int i = 0; i < n; i++) {
+      ifPassed = ifDetailsPresent(editActivitiesList.get(i), articleName, userName);
+      break;
     }
-    if (!ifPassed) {
-      PageObjectLogging.log("verifyRecentEdition",
-                            "edition on article  '" + articleName + "' by user: '"
-                            + userName + "', not found", false, driver);
+
+    return ifPassed;
+  }
+
+  public Boolean doesLastNRecentActivitiesContain(int n, String articleName, String userName) {
+    Boolean ifPassed = false;
+    for (int i = 0; i < n; i++) {
+      ifPassed = ifDetailsPresent(activitiesList.get(i), articleName, userName);
+      break;
+    }
+
+    return ifPassed;
+  }
+
+  public Boolean doesLastNRecentBlogActivitiesContain(int n, String blogPostContent, String blogPostName,
+                                                      String userName) {
+    Boolean ifPassed = false;
+    for (int i = 0; i < n; i++) {
+      ifPassed = ifNewBlogDetailsPresent(activitiesList.get(i), blogPostContent, blogPostName, userName);
+      if (ifPassed == true) break;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Missing curly brace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS00121)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS00122)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the literal "true" boolean value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1125)